docs/administrators/authorization.rst

```diff
@@ -0,0 +1,275 @@
+
+
+Access policy
+"""""""""""""
+
+The access policy is written in the `Rego query language <https://www.openpolicyagent.org/docs/policy-language>`_. The policy must contain a rule named ``allow`` that evaluates to a boolean value. This rule is evaluated for each API call. If the rule evaluates to True, the API call is authorized. Otherwise it's not. The snippet below provides a short policy that grants read-only access to users having the ``read-only`` role and any access to users with the ``user`` role.
+
+.. code-block:: rego
+
+    package policy
+
+    # Get the metadata for the specific endpoint that is called.
+    endpoint_data := data.endpoints[input.request.endpoint_id]
+
+    # Don't allow anything that is not explicitly allowed.
+    default allow := false
+
+    # Give read-only access to users with the read-only role.
+    allow if {
+        input.token.role == "read-only"
+        endpoint_data.read_only == true
+    }
+
+    # Users with the user role are allowed to call any API endpoint.
+    allow if {
+        input.token.role == "user"
+    }
+
+
+Default access policy
+^^^^^^^^^^^^^^^^^^^^^
+
+The Inmanta server comes with a default policy that is defined in ``/etc/inmanta/authorization/policy.rego``. The policy assumes that there are two types of roles: environment-scoped roles and global roles. As the same suggests, environment-scoped roles are only relevant within a specific environment. Global roles are relevant on the entire Inmanta server. The default policy defines the following roles:
+
+1. Environment-scoped roles:
+
+   * **read-only**: Users with this role have read-only access on everything in a certain environment.
+   * **noc**: Users with this role can do anything in a certain environment, except for modifying the desired state, changing environment settings and expert operations.
+   * **operator**: Users with this role can create, update and delete service instances in a certain environment next to all the actions allowed by the noc role.
+   * **environment-admin**: Users with this role can do anything in a certain environment, except for expert operations.
+   * **environment-expert-admin**: Users with this role can do anything in a certain environment.
+
+2. Global admin role: A user with this role can execute any API endpoint on the Inmanta server.
+
+The default policy makes the following assumptions about the content of the access token:
+
+* The ``sub`` claim is present and contains the username of the user the token is for.
+* Environment-scoped roles are defined in the ``urn:inmanta:roles`` claim of the access token. The value must be a dictionary that maps the ID of the environment to a list of roles the user has in that environment.
+* Global admins must have the claim ``urn:inmanta:is_admin`` in the access token with the value set to ``true``.
```

Comment:
Is this relevant to the end user? This is something the server constructs, isn't it?

If put in another subsection, in the light of "this is what you may assume when writing an access policy", I could perhaps see its value, but here it feels off to me.

Reply:
After reading on I see how it's relevant. I think it would be valuable to cross-link the two:
- Here something like "... about the content of the access token. This has to be configured correctly when using third party authorization providers:"
- In the section about enabling the policy engine, in point 3, something like "see <this section> for the claims required by the default policy."